tests/transforms/test_broadcast_expand.py

```diff
@@ -260,43 +233,44 @@ def cost(*params):
 
-        if len(obs) > 1 and size == 1:
-            new_exp_jac = [qml.math.flatten(el) for el in exp_jac]
-            new_jac = [qml.math.flatten(el) for el in jac]
-            assert all(qml.math.allclose(_jac, e_jac) for _jac, e_jac in zip(new_jac, new_exp_jac))
-        else:
-            for _jac, e_jac in zip(jac, exp_jac):
-                if e_jac is None:
-                    assert qml.math.allclose(_jac, 0.0)
-                else:
-                    assert qml.math.allclose(_jac, e_jac)
+        for _jac, e_jac in zip(jac, exp_jac):
+            if e_jac is None:
+                assert qml.math.allclose(_jac, 0.0)
+            else:
+                assert qml.math.allclose(_jac, e_jac)
 
     @pytest.mark.torch
     @pytest.mark.filterwarnings("ignore:Output seems independent of input")
-    @pytest.mark.parametrize("params, size", parameters_and_size)
+    @pytest.mark.parametrize("params", parameters)
     @pytest.mark.parametrize("obs, exp_fn", observables_and_exp_fns)
-    def test_torch(self, params, size, obs, exp_fn):
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    def test_torch(self, params, obs, exp_fn, diff_method):
         """Test that the expansion works with torch and is differentiable."""
         import torch
 
-        torch_params = tuple(torch.tensor(p, requires_grad=True) for p in params)
+        torch_params = tuple(
+            torch.tensor(p, requires_grad=True, dtype=torch.float64) for p in params
+        )
         params = tuple(pnp.array(p, requires_grad=True) for p in params)
 
+        @qml.transforms.broadcast_expand
+        @qml.qnode(dev, interface="torch", diff_method=diff_method)
         def cost(*params):
-            tape = make_tape(*params, obs)
-            tapes, fn = qml.transforms.broadcast_expand(tape)
-            if len(tape.measurements) > 1:
-                return qml.math.stack(fn(qml.execute(tapes, dev, qml.gradients.param_shift)))
-            return fn(qml.execute(tapes, dev, qml.gradients.param_shift))
+            make_ops(*params)
+            return tuple(qml.expval(ob) for ob in obs)
 
+        res = cost(*torch_params)
         jac = torch.autograd.functional.jacobian(cost, torch_params)
-        exp_jac = qml.jacobian(exp_fn)(*params)
+        exp_jac = torch.autograd.functional.jacobian(exp_fn, torch_params)
 
-        if len(obs) > 1 and size == 1:
-            assert qml.math.allclose(cost(*torch_params), exp_fn(*params).T)
-            new_exp_jac = [qml.math.flatten(el) for el in exp_jac]
-            new_jac = [qml.math.flatten(el) for el in jac]
-            assert all(qml.math.allclose(_jac, e_jac) for _jac, e_jac in zip(new_jac, new_exp_jac))
+        if len(obs) > 1:
+            assert all(qml.math.allclose(r, e) for r, e in zip(res, exp_fn(*params)))
+            # Need to perform a transpose because the broadcast_expand transform pulls out the
+            # broadcasting axis, which the `exp_fn` does not do.
```

Comment:
As far as I can tell, the reason for this is that the `cost` function above returns a tuple, whereas the `exp_fn` for `len(obs)>1` returns an array.